test/e2e/aks.go

```diff
@@ -0,0 +1,69 @@
+//go:build e2e
+// +build e2e
+
+package e2e
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"context"
+	"time"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+
+	"github.com/Azure/ARO-RP/pkg/util/liveconfig"
+)
+
+// Tests the kubeconfig ability to get and manipulate the cluster
+var _ = Describe("AKS cluster present", func() {
+	ctx := context.Background()
+	var liveConfig liveconfig.Manager
+	var kubeConfig *rest.Config
+
+	BeforeEach(func() {
+		var err error
+		liveConfig, err = _env.NewLiveConfigManager(ctx)
+
+		Expect(err).To(BeNil())
+	})
+
+	It("should get kubeconfig", func() {
+		var err error
+
+		kubeConfig, err = liveConfig.HiveRestConfig(ctx, 0)
+		Expect(err).To(BeNil())
+		Expect(kubeConfig).ToNot(BeNil())
+
+		kubernetescli, err := kubernetes.NewForConfig(kubeConfig)
+		Expect(err).To(BeNil())
+
+		// to avoid name collision by accident
+		testNamespaceName := "e2e-test-namespace" + time.Now().Format("20060102150405")
```

Comment:
non-blocking nit:
```suggestion
		testNamespaceName := "e2e-test-namespace-" + time.Now().Format("20060102150405")
```

Reply:
We could also use `GinkgoParallelProcess()` to append the process number such that if there happens to be multiple functions using this generic namespace name at the same time, they won't step on each other's toes.

Reply:
Unfortunately the `GinkgoParallelProcess()` will not help. The AKS/Hive is shared between different pipelines. It can easily happen that multiple pipelines will have all process `1`. This was the motivation for selecting date.